#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix mn_MN, sid_ET and vi_VN localedef compilation breakage.
# DP: Related bugs: #321634: locales: vi_VN locale generation fails for lack of LC_NAME, LC_ADDRESS, LC_TELEPHONE
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
# DP: Patch author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Will be submitted
# DP: Status Details: 
# DP: Date: 2005-08-22

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2005-08-22  GOTO Masanori  <gotom@debian.or.jp>

	* locales/mn_MN: Fix invalid line separation.
	* locales/sid_ET: Don't use undefined lang_ab.
	* locales/vi_VN: Add default LC_NAME, LC_ADDRESS, LC_TELEPHONE.

Index: localedata/locales/mn_MN
===================================================================
RCS file: /cvs/glibc/libc/localedata/locales/mn_MN,v
retrieving revision 1.5
diff -u -p -r1.5 mn_MN
--- localedata/locales/mn_MN	22 May 2005 16:20:17 -0000	1.5
+++ localedata/locales/mn_MN	22 Aug 2005 10:13:00 -0000
@@ -218,8 +218,8 @@ am_pm   "";""
 t_fmt_ampm ""
 % Full date and time representation
 % "%Z %Y ony %B %e, %a %H:%M:%S"
-date_fmt  "<U0025><U005A><U0020><U0025><U0059><U0020><U043E><U043D><U044B><U0020><U0025><U0042><U0020>";/
-	  "<U0025><U0065><U002C><U0020><U0025><U0061><U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053>"
+date_fmt  "<U0025><U005A><U0020><U0025><U0059><U0020><U043E><U043D><U044B><U0020><U0025><U0042><U0020>
+<U0025><U0065><U002C><U0020><U0025><U0061><U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053>"
 END LC_TIME
 
 LC_MESSAGES
Index: localedata/locales/sid_ET
===================================================================
RCS file: /cvs/glibc/libc/localedata/locales/sid_ET,v
retrieving revision 1.3
diff -u -p -r1.3 sid_ET
--- localedata/locales/sid_ET	3 May 2004 20:52:23 -0000	1.3
+++ localedata/locales/sid_ET	22 Aug 2005 10:22:21 -0000
@@ -113,7 +113,7 @@ country_num   231   % 210 found in at le
 % country_car   unknown
 % country_isbn  unknown, Need ISO 2108
 lang_name     "<U0053><U0069><U0064><U0061><U0061><U006D><U0075><U0020><U0041><U0066><U006F>"
-lang_ab       "<U0073><U0069><U0064>"
+lang_ab       ""
 lang_term     "<U0073><U0069><U0064>"
 lang_lib      "<U0073><U0069><U0064>"
 %
Index: localedata/locales/vi_VN
===================================================================
RCS file: /cvs/glibc/libc/localedata/locales/vi_VN,v
retrieving revision 1.7
diff -u -p -r1.7 vi_VN
--- localedata/locales/vi_VN	8 Jul 2005 16:19:07 -0000	1.7
+++ localedata/locales/vi_VN	22 Aug 2005 10:44:49 -0000
@@ -283,6 +283,18 @@ height      297
 width       210
 END LC_PAPER
 
+LC_NAME
+copy "i18n"
+END LC_NAME
+
+LC_ADDRESS
+copy "i18n"
+END LC_ADDRESS
+
+LC_TELEPHONE
+copy "i18n"
+END LC_TELEPHONE
+
 LC_MEASUREMENT
 measurement 1
 END LC_MEASUREMENT
